packages/arb-token-bridge-ui/src/hooks/useArbQueryParams.ts

```diff
@@ -36,15 +41,14 @@ export const useArbQueryParams = () => {
 // Defined here so that components can directly rely on this for clean amount values and not rewrite parsing logic everywhere it gets used
 const AmountQueryParam = {
   encode: (amount: number | string | null | undefined) => {
-    // if (amount === 'MAX') return 'MAX' // commenting out, handling for a future case
     if (typeof amount === 'number') return amount.toString()
     return amount
   },
   decode: (amountStr: string | (string | null)[] | null | undefined) => {
-    // if (amountStr === 'MAX') return 'MAX' // commenting out, handling for a future case
     if (
       !amountStr ||
-      isNaN(Number(amountStr)) ||
+      (isNaN(Number(amountStr)) &&
```

Comment:
Tried the experience. From a UX perspective, does it make sense to encode and persist MAX into the URL as well?
Currently `MAX` is only initializing the form but after that, the URL gets overridden by the actual decimal value of the amount.

What do you suggest?

Reply:
I prefer treating `max` as an alias for user to indicate the amount in the URL
The replacement is expected in this case

open to suggestions but I don't think keeping `max` is a better UX

Reply:
That would be hard to do, as max is calculated based on the current gas price and. Let's just keep it for initially setting it.

Reply:
Should we maybe split this into some early returns for readability?

```
  decode: (amountStr: string | (string | null)[] | null | undefined) => {
    if (!amountStr) {
      return ''
    }
    
    if (typeof amountStr === 'object') {
      return ''
    }
    
    if (isNaN(Number(amountStr)) {
      if (amountStr.toLowerCase() === 'max') {
        return 'max'
      }
      
      return ''
    }

    return amountStr
  }
```

Also, curious why we need the `typeof amountStr === 'object'` check?

Reply:
not entirely sure with the object check, @dewanshparashar can we remove it?

Reply:
No, we shouldn't remove it. This check is for values of the format `?amount=abc,def,...` in query params which decodes to an array format. We want the amount to be only a single string, hence this check.